analyzer/java/com/engflow/bazel/invocation/analyzer/dataproviders/LocalActions.java

```diff
@@ -175,6 +187,24 @@ public boolean isRemoteCacheHit() {
       return remoteCacheHit;
     }
 
+    /**
+     * Returns whether the action looks like it is Bazel-internal.
+     *
+     * @return true if it is likely internal, false if it is not, and empty if it is unclear
```

Comment:
Since we treat an "Optional.empty" result as "false" at the call site, we might as well simplify the function signature and return a boolean.

IIUC neither the true nor the false answer is definite. Which is OK -- we make a best guess.

WDYT?

Reply:
I figured with an `Optional` it was clearer that we are "sure" about some, and "don't know" about others. Also, there may be a use case where you want to interpret `empty` as `true`, not false, e.g. if you want to identify actions that are **definitely** not internal.